tests/functional/pyocf/types/volume.py

```diff
@@ -431,23 +431,29 @@ def get_bytes(self):
 
 class ErrorDevice(Volume):
     def __init__(
-        self, vol, error_sectors: set = None, error_seq_no: dict = None, armed=True, uuid=None,
+        self,
+        vol,
+        error_sectors: set = None,
+        error_seq_no: dict = None,
+        data_only=False,
+        armed=True,
+        uuid=None,
     ):
         self.vol = vol
         super().__init__(uuid)
         self.error_sectors = error_sectors
         self.error_seq_no = error_seq_no
+        self.data_only = data_only
         self.armed = armed
         self.io_seq_no = {IoDir.WRITE: 0, IoDir.READ: 0}
         self.error = False
 
     def set_mapping(self, error_sectors: set):
         self.error_sectors = error_sectors
 
-    def do_submit_io(self, io):
+    def _inject_error(self, io):
         if not self.armed:
-            self.vol.do_submit_io(io)
-            return
+            return False
```

Comment:
We should rather do this check outside this function.

Reply:
Done